content/en/docs/distributions/gke/customizing-gke.md

```diff
@@ -14,16 +14,12 @@ Kubernetes Engine (GKE) on Google Cloud.
 
 ## Customizing Kubeflow before deployment
 
-The Kubeflow deployment process is divided into two steps, **build** and 
+The Kubeflow deployment process is divided into two steps, **hydrate** and 
 **apply**, so that you can modify your configuration before deploying your 
 Kubeflow cluster.
 
-Follow the guide to [deploying Kubeflow on Google Cloud](/docs/gke/deploy/deploy-cli/).
-When you reach the 
-[setup and deploy step](/docs/gke/deploy/deploy-cli/#deploy-kubeflow),
-**skip the `kfctl apply` command** and run the **`kfctl build`** command 
-instead, as  described in that step. Now you can edit the configuration files 
-before deploying Kubeflow.
+Follow the guide to [deploying Kubeflow on Google Cloud](/docs/gke/deploy/deploy-cli/). You can add your patches in corresponding component folder, and include those patches in `kustomization.yaml` file. Learn more about the usage of [kustomize](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/). You can also find the exisitng kustomization in [kubeflow/gcp-blueprints](https://github.com/kubeflow/gcp-blueprints) as example. After adding the patches, you can run `make hydrate` to validate the resulting resources. Finally, you can run `make apply` to deploy the customized Kubeflow.
```

Comment:
For discussion: I don't think you need to address these comments for this release.

If they add patches in `kustomization.yaml` directly, how do they upgrade?
my current thoughts to support upgrade is -- when user needs customizations, they move gcp manifests to a subfolder and write their own kustomization.yaml there instead, for example:

```
cd apps/pipelines
mkdir gcp
mv ./* gcp/
# build user customizations in apps/pipelines, including Makefile, kustomization.yaml etc
```

1 month later, user wants to upgrade pipelines:

```
kpt pkg update apps/pipelines/gcp <latest-repo-url>
```

In this way, upgrade and customization do not conflict.

Tradeoff: this approach needs a bit more manual work for upgrade, it's not worth it when customization is simple and can easily be upgraded and resolve merge conflicts.

To save the folder move step, we might consider creating a `gcp` folder upfront for every app/common, and keep minimum wrapper outside of it, but this seems like pre-mature optimization. Asking people who need to customize to move their folders seem more reasonable.

Reply:
Thank you Yuan for the detailed discussion and thoughtful idea!

I agree that it might be a bit overhead for the need of moving folders. Especially when it comes to mismatch of file location between local and remote repos.

It is also a good point of creating a `gcp` folder, but it mean creating a lot fo `gcp` folders. And creating difficulty when it comes to Kubeflow verion upgrade.

I want to add one idea to this discussion:

We provide a simple kustomization.yaml file to allow users to edit, initially the content in this file references to our gcp-blueprints resource using `remote URL`: https://github.com/kubernetes-sigs/kustomize/blob/master/examples/remoteBuild.md#url-format. This approach makes sure that we don't need to change the kustomization.yaml file on the user side, so they can use it for their own customization. During every upgrade, they just need to change the branch name in the remote URL, and evaluate the result in `build` folder. If they want to pull the manifest from remote, they can do so by following our instruction, but by default their local directory is lightweight and contains only their customization.

However, this will be a large refactoring as well. So I agree that we need more discussion around it.

Reply:
discussed the idea over gvc, it's against best practices for using off-the-shelf manifests with kustomize

Reply:
Agreed, let's direct user for individual component upgrade, and figure out future plan for full kubeflow batch upgrade later.